runtime/inspector/mod.rs

```diff
@@ -0,0 +1,722 @@
+    maybe_call_id: Option<i32>,
+    msg: v8::UniquePtr<v8::inspector::StringBuffer>,
+  ) {
+    let msg = msg.unwrap().string().to_string();
+    let _ = self.proxy_tx.unbounded_send((maybe_call_id, msg));
+  }
+
+  pub fn break_on_next_statement(&mut self) {
+    let reason = v8::inspector::StringView::from(&b"debugCommand"[..]);
+    let detail = v8::inspector::StringView::empty();
+    self
+      .v8_session
+      .borrow_mut()
+      .as_mut()
+      .unwrap()
+      .schedule_pause_on_next_statement(reason, detail);
+  }
+}
+
+impl v8::inspector::ChannelImpl for WebsocketSession {
+  fn base(&self) -> &v8::inspector::ChannelBase {
+    &self.v8_channel
+  }
+
+  fn base_mut(&mut self) -> &mut v8::inspector::ChannelBase {
+    &mut self.v8_channel
+  }
+
+  fn send_response(
+    &mut self,
+    call_id: i32,
+    message: v8::UniquePtr<v8::inspector::StringBuffer>,
+  ) {
+    self.send_message(Some(call_id), message);
+  }
+
+  fn send_notification(
+    &mut self,
+    message: v8::UniquePtr<v8::inspector::StringBuffer>,
+  ) {
+    self.send_message(None, message);
+  }
+
+  fn flush_protocol_notifications(&mut self) {}
+}
+
+impl Future for WebsocketSession {
+  type Output = ();
+  fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
+    self.proxy_rx_handler.poll_unpin(cx)
+  }
+}
+
+/// A local inspector session that can be used to send and receive protocol messages directly on
+/// the same thread as an isolate.
+pub struct InMemorySession {
+  v8_channel: v8::inspector::ChannelBase,
+  v8_session: v8::UniqueRef<v8::inspector::V8InspectorSession>,
+  response_tx_map: HashMap<i32, oneshot::Sender<serde_json::Value>>,
+  next_message_id: i32,
+  notification_queue: Vec<Value>,
+}
+
+impl v8::inspector::ChannelImpl for InMemorySession {
+  fn base(&self) -> &v8::inspector::ChannelBase {
+    &self.v8_channel
+  }
+
+  fn base_mut(&mut self) -> &mut v8::inspector::ChannelBase {
+    &mut self.v8_channel
+  }
+
+  fn send_response(
+    &mut self,
+    call_id: i32,
+    message: v8::UniquePtr<v8::inspector::StringBuffer>,
+  ) {
+    let raw_message = message.unwrap().string().to_string();
+    let message: serde_json::Value = match serde_json::from_str(&raw_message) {
+      Ok(v) => v,
+      Err(error) => match error.classify() {
+        serde_json::error::Category::Syntax => json!({
+          "id": call_id,
+          "result": {
+            "result": {
+              "type": "error",
+              "description": "Unterminated string literal",
+              "value": "Unterminated string literal",
+            },
+            "exceptionDetails": {
+              "exceptionId": 0,
+              "text": "Unterminated string literal",
+              "lineNumber": 0,
+              "columnNumber": 0
+            },
+          },
+        }),
+        _ => panic!("Could not parse inspector message"),
+      },
+    };
+
+    self
+      .response_tx_map
+      .remove(&call_id)
+      .unwrap()
```

Comment:
Note that unwrapping here can cause the deno process to crash if the client sends a response with an invalid call_id - this might be a problem for Deploy.

Reply:
I'm aware of this problem - for now this is just as it was implemented originally - there were no changes to `InMemorySession` besides renaming it